apps/core/src/hooks/useTransactionSummary.ts

```diff
@@ -45,6 +50,140 @@ const getSummary = (
     }
 };
 
+const getObjectSummaryNFTData = <T extends { objectId: string }>({
+    nftsMeta,
+    originalData,
+}: {
+    nftsMeta: ReturnType<typeof useNFTsMeta>;
+    originalData: T[];
+}) => {
+    const objectIdsFromNFTData = nftsMeta.data.objectIds;
+    const objectIdsFromNFTDataSet = new Set(objectIdsFromNFTData);
+
+    return originalData.reduce((acc, obj) => {
+        if (objectIdsFromNFTDataSet.has(obj.objectId)) {
+            const index = nftsMeta.data.objectIds.indexOf(obj.objectId);
+
+            return [
+                ...acc,
+                {
+                    ...obj,
+                    nftMeta: nftsMeta.data.metaData[index],
+                },
+            ];
+        }
+        return acc;
+    }, [] as (T & { nftMeta: Record<string, string | null> })[]);
+};
+
+export function useTransactionSummaryWithNFTs({
```

Comment:
Why do we have two hooks? In what case do we not want the NFTS? If there is one can't we just make this an option passed to the other hook?

Reply:
The wallet is still using the other one and the output is pretty different (array vs objects, etc). Ideally, we should be using this new hook instead since it has more grouping logic in there. We should deprecate the other one after the wallet is caught up with the current implementation in explorer.

Reply:
I'm also working on updating the summary hook with grouping logic / display data. let's set up some time to sync on this